### DISCUSSION TITLE
Handling toxic users

### DISCUSSION THREAD

**Discussion by twpayne** [twpayne/chezmoi#2819] [Maintainance]
The vast majority of chezmoi users are happily using the software, and a few are contributing ideas for improvements, reporting bugs, and even helping other users, which is fantastic. By simple statistics, as the number of users grows, so does the absolute number of toxic users. Toxic users have a extremely detrimental effect on maintainers and can lead to maintainers burning out and abandoning the project if they are not handled quickly decisively.

In short, chezmoi is free software made available to everybody to use as they wish. Maintainers and contributors have zero obligations to users. When you encounter a toxic user please, for your own mental health, do not engage with them.

Some insightful articles:
* "Did You Miss My Comments Or What?” Toxicity In Open Source Discussions" [summary](https://tylercipriani.com/blog/2023/01/15/entitlement-burnout-and-toxicity/) and [paper](https://cmustrudel.github.io/papers/osstoxicity22.pdf).
* [There is no "software supply chain"](https://iliana.fyi/blog/software-supply-chain/)
* [Entitlement in Open Source](https://mikemcquaid.com/entitlement-in-open-source/)
* [Open Source Maintainers Owe You Nothing](https://mikemcquaid.com/open-source-maintainers-owe-you-nothing/)

**Comment by Lockszmith-GH** [comment]
@twpayne I'm assuming your post is following some toxic interaction - if that is the case, sending you (or any maintainer who dealt with that) words of support from myself and on behalf of the small group of users I personally have introduced to chezmoi.

The care, thought, work you and the other maintainers put in, as well as the community you have cultivated around this tool is absolutely amazing.

And to any potential troll out there, shame on you.

**Comment by twpayne** [comment]
The article [How ‘open’ should your open source be?](https://github.com/readme/featured/how-open-is-open-source) on the GitHub blog is well worth reading.

**Reply by Lockszmith-GH** [reply]
Thanks for sharing. Interesting read.

As a FOSS enthusiast myself, when talking with other techs who have the mentality of "it's free, so I take it" - I find myself explaining that a successful project isn't just about the code. A quote in the article states this best, when Julia Ferraioli is quoted:
> “As soon as you get noticed, you essentially have an unlimited number of stakeholders,” says Ferraioli. “You’re suddenly plunged into project management, diplomacy, marketing, and branding.”

The article describes how the PR was a feature that drove the ease of participating with FOSS projects, I think the next big step was introducing Discussions as part of the project.
Although I always fear the creation of additional channels of communication for myself, as we're all already overwhelmed by streams of input.

I find you utilize these tools here very well, and this is one of the projects that I cherish my participation in - even if it is mostly here in the discussions.

How do you deal with this @twpayne ? Has this project evolved into what you envisioned originally when you started it?

**Reply by twpayne** [reply]
Thanks for the kind words! Certainly, chezmoi has grown much larger in terms of both users and features than I ever expected.

To keep the support burden manageable, I'm aiming for as few bugs as possible and a lot of documentation. Any bug will be hit by multiple users and so by fixing bugs quickly hopefully it is only necessary to deal with one bug report. Good documentation should reduce the number of support requests.

**Reply by Lockszmith-GH** [reply]
You've done a phenomenal job of building the documentation and process around its development, that much I've seen.
As a DevOps engineer (I consider myself somewhat decent), I am impressed (and probably somewhat envious).

Documentation is great, but... alas not everyone starts with reading the docs. That's why I like helping out in a community that cherishes the docs - pointing to the docs is something most people do here.

Out of curiosity, how much time out of the day does chezmoi take out of you? (when you're not actively developing a new feature/concept)

**Reply by twpayne** [reply]
> Out of curiosity, how much time out of the day does chezmoi take out of you?

Oops, I realize that I never answered your question.

It's about half a day per week. I'm insanely grateful to chezmoi's many maintainers and contributors.

**Comment by cig0** [comment]
I came here to ask for help and found this discussion.

Instead, this is my love letter to Chezmoi:

Until a couple of months ago, I was still managing my dotfiles using a combination of symlinks + Git heavily inspired by GNU Stow. For the past two years, it served me well to keep my hosts in sync. It served me so well that I was reluctant to spend time trying Chezmoi; after all: how much could Chezmoi improve my workflow?

![image](https://github.com/twpayne/chezmoi/assets/394089/4748a080-e442-4ec6-bb4b-d5c32ce2a521)

_An accurate depiction of myself managing my dotfiles with my old homebrew solution_

The funny thing is that I came to know about Chezmoi after learning that Home Manager could help me better manage my dotfiles, not only my Nix packages. So I played a little with Nix Flakes and liked what I saw. I was %99,9 sure I was ready to start working on the new stuff, but me being me, I also decided to give Chezmoi a try.
I can hardly find the right words to summarize the positive change Chezmoi made in my life (partly because English is not my first language, heh). I'm not exaggerating here: I spend some good 10~12 hours doing _computer stuff_ so having such a potent and friendly tool as Chezmoi is a game changer for me.

Right off the bat, the ability to GPG-encrypt files and directories on-the-fly before uploading them is a chef's kiss. Being able to make some changes, push them to a remote private repository then equally easily grab those changes everywhere else? The templating system? The ability to add/remove/forget/ignore entities? Commands like `managed`, `unmanaged`, `edit`, `state`, `purge`, and `doctor`, to name some? Beyond cool.

I was barely into my first week of using Chezmoi when I knew I wasn't looking back to anything else. Chezmoi felt right at home (no pun intended) from the get-go. And I believe that's a hallmark of a well-designed, well-crafted tool, right? You know something is good (anything) when there's no friction in its adoption. It's a good change of pace that, for once, I don't feel I have to fight with a tool, but instead, it behaves like a good old friend, and you can have fun with it.

Thank you! :raised_hands:

**Reply by twpayne** [reply]
Thank you @cig0 <3

**Reply by airtonix** [reply]
you should checkout comtrya then.

**Comment by lnyousif** [comment]
As I tried to figure out the best way to handle moving across many machines, Chezmoi proved to be amazingly useful. and as I search for solution to common issues i face, I always see @twpayne engaged and explain in details how to manage it. I would like to extend my gratitude and thanks for all the work you all do. The entitlement surround us unfortunately, but always remember that the number of people who appreciate your brilliance is far ahead of few entitled noise makers.
